service/src/main/java/datawave/microservice/authorization/OAuthOperations.java

```diff
@@ -0,0 +1,203 @@
+package datawave.microservice.authorization;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import datawave.microservice.authorization.oauth.*;
+import datawave.microservice.authorization.user.ProxiedUserDetails;
+import datawave.security.authorization.*;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.swagger.annotations.ApiOperation;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Presents the REST operations for the authorization service.
+ */
+@RestController
+@RequestMapping(path = "/v1/oauth", produces = MediaType.APPLICATION_JSON_VALUE)
+public class OAuthOperations {
+    private Logger log = LoggerFactory.getLogger(getClass());
+    private final JWTTokenHandler tokenHandler;
+    private final CachedDatawaveUserService cachedDatawaveUserService;
+    private final OAuthProperties oAuthProperties;
+    
+    private Cache<String,AuthorizationRequest> CACHE;
+    
+    @Autowired
+    public OAuthOperations(JWTTokenHandler tokenHandler, CachedDatawaveUserService cachedDatawaveUserService, OAuthProperties oAuthProperties) {
+        this.tokenHandler = tokenHandler;
+        this.cachedDatawaveUserService = cachedDatawaveUserService;
+        this.oAuthProperties = oAuthProperties;
+        long authCodeTtl = this.oAuthProperties.getAuthCodeTtl(TimeUnit.SECONDS);
+        if (authCodeTtl == -1) {
+            throw new IllegalStateException("authCodeTtl not configured.");
+        } else {
+            Caffeine<Object,Object> caffeine = Caffeine.newBuilder();
+            caffeine.expireAfterWrite(authCodeTtl, TimeUnit.SECONDS);
+            CACHE = caffeine.build();
```

Comment:
Is local cache sufficient for this scenario? Wondering if something backed by `@CacheConfig` / `@Cacheable` could be needed here, similar to  our CachedDatawaveUserService impls, for the case when multiple instances of the service are deployed and we need to plug in something like Hazelcast, etc